source/platforms/GitLab.ts

```diff
@@ -162,12 +184,24 @@ class GitLab implements Platform {
     return await this.api.deleteMergeRequestNote(nid)
   }
 
+  /**
+   * Attempts to delete the "main" Danger comment. If the "main" Danger
+   * comment has any comments on it then that comment will not be deleted.
+   */
   deleteMainComment = async (dangerID: string): Promise<boolean> => {
-    //We fetch the discussions even if we are not set to use threads because users could still have replied to a
-    // comment by danger and thus created a discussion/thread. To not leave dangling notes, we delete the entire thread.
-    //There is no API to delete entire discussion. They can only be deleted fully by deleting every note:
-    const discussions = await this.getDangerDiscussions(dangerID)
-    return await this.deleteNotes(this.reduceNotesFromDiscussions(discussions))
+    if (useThreads()) {
+      //There is no API to delete entire discussion. They can only be deleted fully by deleting every note:
+      const discussions = await this.getDangerDiscussions(dangerID)
+      return await this.deleteNotes(this.reduceNotesFromDiscussions(discussions))
+    } else {
+      const notes = await this.getMainDangerNotes(dangerID)
```

Comment:
This is copied from the code that existed before the change in https://github.com/danger/danger-js/pull/1331